components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java

```diff
@@ -89,4 +90,34 @@ public UserSession getSession(String sessionId) throws SessionManagementServerEx
         }
         return null;
     }
+
+    /**
+     * Get federated authentication session mapping info during the federated idp init logout
+     *
+     * @param fedIdpSessionId
+     * @return A FederatedUserSession containing federated authentication session details
+     * @throws SessionManagementServerException
+     */
+    public FederatedUserSession getFederatedAuthSessionDetails(String fedIdpSessionId)
+            throws SessionManagementServerException {
+
+        FederatedUserSession federatedUserSession;
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        try {
+            federatedUserSession = jdbcTemplate
+                    .fetchSingleRecord(SQLQueries.SQL_GET_FEDERATED_AUTH_SESSION_INFO_BY_SESSION_ID,
+                            (resultSet, rowNumber) -> new FederatedUserSession(
+                                    resultSet.getString(SessionMgtConstants.FEDERATED_IDP_SESSION_ID),
+                                    resultSet.getString(SessionMgtConstants.FEDERATED_SESSION_ID),
+                                    resultSet.getString(SessionMgtConstants.FEDERATED_IDP_NAME),
+                                    resultSet.getString(SessionMgtConstants.FEDERATED_AUTHENTICATOR_ID),
+                                    resultSet.getString(SessionMgtConstants.FEDERATED_PROTOCOL_TYPE)),
+                            preparedStatement -> preparedStatement.setString(1, fedIdpSessionId));
+            return federatedUserSession;
+        } catch (DataAccessException e) {
+            throw new SessionManagementServerException(
+                    SessionMgtConstants.ErrorMessages.ERROR_CODE_UNABLE_TO_GET_SESSION,
```

Comment:
Shall we add a new error message and error code for getting federated user session? 
ERROR_CODE_UNABLE_TO_GET__FED_USER_SESSION